src/tree/contexts/ContextTreeItem.ts

```diff
@@ -60,6 +61,18 @@ export class ContextTreeItem extends AzExtTreeItem {
     }
 
     public async use(context: IActionContext): Promise<void> {
-        return dockerContextManager.use(this.name);
+        try {
+            ext.isContextSwitchInProgress = true;
+            return this.runWithTemporaryDescription(
```

Comment:
Does this actually run the `finally` in the expected order since we are not `await`ing it here?

Reply:
Excellent catch @bwateratmsft ! I bet the `finally` clause needs to be inside the lambda passed to `runWithTemporaryDescription`

Reply:
I'm not 100% certain that would have the expected behavior, but I am 100% that `await this.runWithTemporaryDescription(...)` and then `finally` would. If `runWithTemporaryDescription` threw an exception for whatever reason, we would not hit a `finally` inside the lambda.

Reply:
Good catch!